### ISSUE TITLE
Command Injection in append_to_squashed_commit_message.

### ISSUE THREAD

**Issue by dreid** [reenhanced/gitreflow#134]
https://github.com/reenhanced/gitreflow/blob/master/lib/git_reflow/git_helpers.rb#L64

The message is interpolated into this command sequence causing variable and command substitution to happen in the shell.

Most notably this causes commands in `backticks` found in the PR description to be executed upon `git reflow deliver`.

My suggestion in this particular case would be to just use the File api to construct the SQUASH_MSG file and avoid the shell entirely.

However I'd also recommend auditing other uses of `run`. 

http://gavinmiller.io/2015/fixing-command-injection-vulnerabilities/ seems to be a pretty good resource.

**Comment by josephgrossberg** [comment]
Helpful context IMO: I often use the backticks, so that my commit messages have fixed-width text on github, per github-flavored mardown. E.g. "adds auditing to the `ShoppingCart` model`"

**Comment by dreid** [comment]
I do not believe this is actually fixed by #135.

My understanding is that `%x{}` works just like backtick treating it's arguments as a string and doing shell expansion.

Take the following irb session for example:

```
irb(main):024:0> user_input = "`date`"
=> "`date`"
irb(main):025:0> `echo "hello, #{user_input}"`
=> "hello, Mon Sep 28 11:39:16 PDT 2015\n"
irb(main):026:0> %x{echo "hello, #{user_input}"}
=> "hello, Mon Sep 28 11:39:23 PDT 2015\n"
irb(main):027:0> system("echo", "hello, #{user_input}")
hello, `date`
=> true
```

As you can see, only the third form (system w/ multiple arguments) can adequately avoid shell substitution.

**Comment by codenamev** [comment]
I'll have to revisit this tomorrow morning.  It appears that `%x{}` does shell expansion only when doing String interpolation via `#{...}` in combination with double quotes.  As an example of it working without it:

```bash
irb(main):001: > user_input = "hello, `date`"
=> "hello, `date`"
irb(main):001: > %x{ echo '#{user_input}' }
=> "hello, `date`"
```

We started using backticks as it gives the most flexibility while maintaining context by avoiding running commands in a subshell (via `system`) or overriding the current process (via `exec`).

I was hoping to get a quick fix in today for you.  I performed an audit on the `run` command and the `append_to_squashed_commit_message` is the only problematic area.  I'll try and get a refactor of that to use Ruby's native file API and will patch this oversight in the morning.
